snapmaker/scripts/pack.py

```diff
@@ -37,11 +37,10 @@ def pack_minor_image(image_type, start_id, end_id, version, input, output):
     else:
         full_version = 'Snapmaker_{}_{}'.format(version, date).encode('UTF-8')
     if len(full_version) >= 32:
-        version = version[:32]
+        full_version = full_version[:32]
     else:
-        for i in range(32 - len(full_version)):
-            full_version += b'\0'
-        full_version += b'\0'
+        full_version += b'\0' * (32 - len(full_version))
+    full_version += b'\0'
```

Comment:
Thank you for your commit.   This line will make the full_version 33 bytes, although the length restriction at 119 lines and will not be a problem, please remove this line for easier understanding, thank you very much.

Reply:
> This line will make the full_version 33 bytes

Right, this ensures that the `full_version` is always 33 bytes long. That's required because the minor image header reserves 33 bytes for the version (see e.g. line 50)

> although the length restriction at 119 lines

This is independent from the restriction in line 119. In line 119 the `full_version` of the *major* image gets restricted to 32 bytes, this restricts the length of the `full_version` of the *minor* image. Since these version strings are independent of one another, both restrictions are necessary.

> please remove this line for easier understanding

If this line gets removed the version has only 32 bytes and therefore the minor image header (which assumes 33 bytes) will be one byte too short and the following elements of the header will appear one byte earlier than expected. Since this affects the checksum and the length field, this would lead to invalid images.

Reply:
I'm sorry for the wrong comment, it's actually 32 bytes, I'll re-modify and submit a [#204](https://github.com/Snapmaker/Snapmaker2-Controller/pull/204) based on this pr, thanks again for your support